_posts/2025-03-13-CPANSec-Retrospective-2024.md

```diff
@@ -0,0 +1,146 @@
+
+## Adopting security related modules
+CPANSec adopted security related modules:
+- [Crypt::DSA](https://metacpan.org/dist/Crypt-DSA)
+- [Crypt::DES_EDE3](https://metacpan.org/dist/Crypt-DES_EDE3)
+- [Crypt::OpenPGP](https://metacpan.org/dist/Crypt-OpenPGP)
+- [Crypt::OpenSSL::AES](https://metacpan.org/dist/Crypt-OpenSSL-AES)
+- [Crypt::OpenSSL::Blowfish](https://metacpan.org/dist/Crypt-OpenSSL-Blowfish)
+- [Crypt::OpenSSL::ConfiguredAPI](https://metacpan.org/dist/Crypt-OpenSSL-ConfiguredAPI)
+- [Crypt::OpenSSL::PKCS10](https://metacpan.org/dist/Crypt-OpenSSL-PKCS10)
+- [Crypt::OpenSSL::SignCSR](https://metacpan.org/dist/Crypt-OpenSSL-SignCSR)
+- [Crypt::OpenSSL::Verify](https://metacpan.org/dist/Crypt-OpenSSL-Verify)
+- [Crypt::OpenSSL::VerifyX509](https://metacpan.org/dist/Crypt-OpenSSL-VerifyX509)
+- [Crypt::Primes](https://metacpan.org/dist/Crypt-Primes)
+- [Crypt::Random](https://metacpan.org/dist/Crypt-Random)
+- [Module::Signature](https://metacpan.org/dist/Module-Signature)
+  
+Adopting modules is not always the only option and CPANSec also reviewed modules for unsecure algorithms
+in order to make recommendations or push some modules to deprecation.
+
+## Reviewing CPAN for supply chain attacks
+Through the year, CPANSec tried to come with analysis of the CPAN ecosystem 
+to know exactly how much vulnerable it would be to some common supply chain attacks:
+
+- [Study CPAN Dependency Confusion](https://security.metacpan.org/docs/cpan-dependency-confusion.html)
+- [Study CPAN StarJacking](https://security.metacpan.org/docs/cpan-starjacking.html)
+
+## Pentesting PAUSE
+CPANSec reviewed and tested PAUSE with various attempts to break it with nasty modules, steal namespaces or execute remote code.
+
+This was done on a local instance that was a good opportunity to work on automating more the reproducibility and ease of setup.
+
+## Other
+CPANSec also opened discussion (with a proposed change) to [Secure by default sessions in Mojolicious](https://github.com/mojolicious/mojo/pull/2200)
+
+Through the year, CPANSec discussed the topic of module signatures. Reviewing the limitations "by design" of the current options and thinking about a correct way to implement that feature.
+Similarly, discussions about CPAN installer enhancements ("Do not install module version with a CVE", "Patch module on the fly at install". etc...) were discussed, but still early stage.
+
+Very end of 2024, CPANSec folks reviewed the state of the art vulnerability of Perl bcrypt related modules ([Crypt::bcrypt](https://metacpan.org/dist/Crypt-Bcrypt), [Digest::Bcrypt](https://metacpan.org/pod/Digest::Bcrypt)) on misuse following the [Okta reported incident](https://medium.com/@rajat29gupta/bcrypt-and-the-okta-incident-what-developers-need-to-know-9d13a446738a).
+
+## Conclusion
+This was a big year, the group really took shape and get full speed with outstanding results.
```

Comment:
One comment (related to the "We" vs. "CPANSec" proposal earlier/above): The text as it is now seems to be written from an outsider's perspective. Would it be better to change this to an insider's perspective?

Reply:
This might mean we'd have to tone down the bragging a little, though :sweat_smile: